crates/paimon/src/spec/schema_change.rs

```diff
@@ -0,0 +1,519 @@
+            field_name: field_names,
+            nullable: new_nullability,
+        }
+    }
+
+    /// impl the `update_column_comment`.
+    pub fn update_column_comment(field_name: String, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment {
+            field_names: vec![field_name],
+            new_description: comment,
+        }
+    }
+
+    /// impl the `update_columns_comment`.
+    pub fn update_columns_comment(field_names: Vec<String>, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment {
+            field_names,
+            new_description: comment,
+        }
+    }
+
+    /// impl the `update_column_position`.
+    pub fn update_column_position(column_move: ColumnMove) -> Self {
+        SchemaChange::UpdateColumnPosition { column_move }
+    }
+}
+
+/// The type of move.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L412>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub enum ColumnMoveType {
+    FIRST,
+    AFTER,
+}
+
+/// Represents a requested column move in a struct.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L410>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ColumnMove {
+    field_name: String,
+    referenced_field_name: Option<String>,
+    #[serde(rename = "type")]
+    move_type: ColumnMoveType,
+}
+
+impl ColumnMove {
+    /// Create a new `Move`.
+    pub fn new(
+        field_name: String,
+        referenced_field_name: Option<String>,
+        move_type: ColumnMoveType,
+    ) -> Self {
+        Self {
+            field_name,
+            referenced_field_name,
+            move_type,
+        }
+    }
+
+    /// Get the field name.
+    pub fn field_name(&self) -> &str {
+        &self.field_name
+    }
+
+    /// Get the referenced field name.
+    pub fn referenced_field_name(&self) -> Option<&str> {
+        self.referenced_field_name.as_deref()
+    }
+
+    /// Get the move type.
+    pub fn move_type(&self) -> &ColumnMoveType {
+        &self.move_type
+    }
+
+    /// Create a new `Move` with `FIRST` move type.
+    pub fn first(field_name: String) -> Self {
+        ColumnMove {
+            field_name,
+            referenced_field_name: None,
+            move_type: ColumnMoveType::FIRST,
+        }
+    }
+
+    /// Create a new `Move` with `AFTER` move type.
+    pub fn after(field_name: String, referenced_field_name: String) -> Self {
+        ColumnMove {
+            field_name,
+            referenced_field_name: Some(referenced_field_name),
+            move_type: ColumnMoveType::AFTER,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::spec::{DataType, IntType};
+
+    #[test]
+    fn test_set_option() {
```

Comment:
Would you like to add a test against real string?

Reply:
> Would you like to add a test against real string?

Are these tests necessary? If they are not, I will move out and add `serialize/deserialize` tests for the `schema_change`.

Reply:
> If they are not, I will move out and add `serialize/deserialize` tests for the `schema_change`.

That makes more sense.